packages/truffle-compile/compilerSupplier/loadingStrategies/Docker.js

```diff
@@ -0,0 +1,86 @@
+const request = require("request-promise");
+const fs = require("fs");
+const { execSync } = require("child_process");
+const LoadingStrategy = require("./LoadingStrategy");
+
+class Docker extends LoadingStrategy {
+  async getSolc(commitString) {
+    const solcFileName = this.getCachedSolcFileName(commitString);
+    if (solcFileName) return this.getCachedSolcByFileName(solcFileName);
+
+    const allVersions = await this.getSolcVersions(this.config.versionsUrl);
+    const fileName = this.getVersionUrlSegment(commitString, allVersions);
+
+    if (!fileName) throw this.errors("noVersion", version);
+
+    const url = this.config.compilerUrlRoot + fileName;
+    const spinner = ora({
+      text: "Downloading compiler",
+      color: "red"
+    }).start();
+
+    return this.getSolcByUrlAndCache(url, fileName, spinner);
+  }
```

Comment:
Can we get rid of this implementation here and instead just construct a `VersionRange` strategy inside the Docker implementation?

Reply:
So if I understand you correctly, the idea would be to make Docker and Native a subcase of VersionRange?

Reply:
Not a subcase, but instead to make them compose the VersionRange behavior.

So inside `load`, something like:
```javascript
const solcjs = new VersionRange().load(userSpecification);

/* ... */
return {
  /* compile, etc. */
  importsParser: solcjs
}
```